src/ControlSystem/ApparentHorizons/Measurements.hpp

```diff
@@ -0,0 +1,125 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <string>
+
+#include "ApparentHorizons/ComputeHorizonVolumeQuantities.hpp"
+#include "ApparentHorizons/Tags.hpp"
+#include "ControlSystem/Protocols/Measurement.hpp"
+#include "ControlSystem/Protocols/Submeasurement.hpp"
+#include "ControlSystem/RunCallbacks.hpp"
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/LinkedMessageQueue.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "ParallelAlgorithms/Interpolation/Callbacks/ErrorOnFailedApparentHorizon.hpp"
+#include "ParallelAlgorithms/Interpolation/Callbacks/FindApparentHorizon.hpp"
+#include "ParallelAlgorithms/Interpolation/Interpolate.hpp"
+#include "ParallelAlgorithms/Interpolation/Targets/ApparentHorizon.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+template <size_t VolumeDim>
+class ElementId;
+template <size_t Dim>
+class Mesh;
+namespace Parallel {
+template <typename Metavariables>
+class GlobalCache;
+}  // namespace Parallel
+namespace domain::Tags {
+template <size_t Dim>
+struct Mesh;
+}  // namespace domain::Tags
+/// \endcond
+
+namespace control_system::ah {
+enum class HorizonLabel { AhA, AhB };
```

Comment:
I like having the `enum class` to represent which horizon we have, but I don't like that it's in the control system library. I think we should have a global identifier (`enum class`) for everything related to horizon A, B, and C (or potentially even objects A,B,C, they don't necessarily have to be horizons). That way there aren't multiple identifiers running around which can be confusing. Then all code that's identical between horizon A and B doesn't have to be duplicated, like interpolation targets, control systems, options, etc.

What do you think?

Reply:
Yeah, this should be more general.  I don't think we have a place for generic two-body stuff yet, so maybe in `src/ApparentHorizons` for now?

Reply:
Yes I think that's a good place.